src/python/pants/backend/rust/target_types.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, MultipleSourcesField, Target
+
+
+class RustCrateSourcesField(MultipleSourcesField):
+    default = ("Cargo.toml", "src/**/*.rs", "tests/**/*.rs")
+    uses_source_roots = False
```

Comment:
So this is pretty tricky to model. We really have two types of sources fields going on - cargo.toml and the normal .RS files. It is very useful for those to be seen as completely separate fields, so that we can do things like be confident every single file is a rust file and it's safe to feed into rustc and rustfmt.

The issue is that we strongly assume there is only one sources field per target type. That is pretty crucial to have the target API Works when you say tgt.get(SourcesField). The safer thing to do would have been to have that return a tuple, but we intentionally return a single field or none.

There is one workaround I can think of, which is to use `SecondaryOwnerMixin` with a dedicated field for the cargo.toml. It is hacky, but I think would work.

--

How do you think we should handle cargo.lock?